src/kOS/Module/kOSVesselModule.cs

```diff
@@ -237,6 +237,15 @@ private void AddDefaultParameters()
         private void HookEvents()
         {
             ConnectivityManager.AddAutopilotHook(parentVessel, UpdateAutopilot);
+
+            // HACK: The following events and their methods are a hack to work around KSP's limitation that
+            // blocks our autopilot from having control of the vessel if out of signal and the setting
+            // "Require Signal for Control" is enabled.  It's very hacky, and my have unexpected results.
+            if (Vessel.vesselType != VesselType.Unknown && Vessel.vesselType != VesselType.SpaceObject)
+            {
+                TimingManager.FixedUpdateAdd(TimingManager.TimingStage.ObscenelyEarly, cacheControllable);
+                TimingManager.FixedUpdateAdd(TimingManager.TimingStage.BetterLateThanNever, resetControllable);
```

Comment:
Yeah... Ironically, the underlying names that they actually use are `Timing0` and `Timing5`.  Just one of those random things Squad has done that looks hacky.